Extensions/Screenshot/screenshottools.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @memberof gdjs
+ * @class screenshot
+ * @static
+ * @private
+ */
+
+gdjs.screenshot = {}
+
+/**
+ * Save a screenshot of the game.
+ * @param {string} savepath The path where to save the screenshot
+ */
+gdjs.screenshot.takeScreenshot = function (savePath, runtimeScene) {
+  let electron = gdjs.RuntimeGamePixiRenderer.getElectron();
+  let canvas = null;
+  let content = null;
+  let fileSystem = null;
+
+  if (electron) {
+    fileSystem = electron.remote.require("fs");
+    canvas = runtimeScene._renderer._pixiRenderer.view;
```

Comment:
So we have to be careful here as renderer might not always be the same.
Always avoid using private variables (that are denoted with an underscore: `_doSomething`. It's a convention to say "this is private, don't use me if you're not the class itself). Fortunately for you instead, there is a function called `getRenderer()` ;)

Even after getting it, you still need to access _pixiRenderer, which is a private member of `gdjs.RuntimeScenePixiRenderer`. This means that I could decide, or another person, to rename it at some point, and break without knowing this extension (because one could think that, as it's private, nobody will be using it outside of the class).

What you want to do instead if to get the renderer from the `gdjs.RuntimeGame` (`runtimeScene.getGame().getRenderer()`) which does have a method `getPIXIRenderer`:

`const canvas = runtimeScene.getGame().getRenderer().getPIXIRenderer();`

Last note, we may actually be running the game and the extension... without PIXI! :D 
The game engine is built around a set of classes that are purely JavaScript and independents of the underlying rendering engine. Instead, the rendering classes, for example the PIXI ones, are stored in files called `xxx-renderingengine-renderer.js`. For example `runtimegame-pixi-renderer.js`.
They are then "injected" in the game engine, which store them inside property into classes that are accessible by `getRenderer()` methods (by convention).

All of this to say that the "thing" returned by getRenderer might actually not be a pixi renderer for a runtime game. So we have to check this:

```
const runtimeGameRenderer = runtimeScene.getGame().getRenderer();
if (typeof gdjs.RuntimeGamePixiRenderer === "function" && runtimeGameRenderer instanceof gdjs.RuntimeGamePixiRenderer) {
  // We're now sure that runtimeGameRenderer is a gdjs.RuntimeGamePixiRenderer.
  // We can even add this "type hint" to get autocompletion!
  /** @type gdjs.RuntimeGamePixiRenderer */
  const canvas = runtimeGameRenderer.getPIXIRenderer();

  const fileSystem = electron.remote.require("fs");
  const content = canvas.toDataURL('image/png').replace('data:image/png;base64,', '');
} else {
  console.error("Screenshot are not supported on this environment/platform.");
}
```